pkg/proxy/winuserspace/proxier.go

```diff
@@ -196,60 +188,64 @@ func (proxier *Proxier) SyncLoop() {
 
 // Ensure that portals exist for all services.
 func (proxier *Proxier) ensurePortals() {
-	proxier.mu.Lock()
+	/*proxier.mu.Lock()
 	defer proxier.mu.Unlock()
 	// NB: This does not remove rules that should not be present.
 	for name, info := range proxier.serviceMap {
 		err := proxier.openPortal(name, info)
 		if err != nil {
 			glog.Errorf("Failed to ensure portal for %q: %v", name, err)
 		}
-	}
+	}*/
 }
 
 // cleanupStaleStickySessions cleans up any stale sticky session records in the hash map.
 func (proxier *Proxier) cleanupStaleStickySessions() {
 	proxier.mu.Lock()
 	defer proxier.mu.Unlock()
-	for name := range proxier.serviceMap {
+	spnMap := make(map[proxy.ServicePortName]bool)
+	for k := range proxier.serviceMap {
+		spn := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: k.Namespace, Name: k.Name}, Port: k.Port}
+		spnMap[spn] = true
+	}
+	for name := range spnMap {
```

Comment:
This indirection seems weird, why not just `proxier.loadBalancer.CleanupStaleStickySessions(name)` where you set `spnMap[spn]=true`